torch/_dynamo/output_graph.py

```diff
@@ -999,6 +994,19 @@ def cleanup_graph(self):
                     self.graph.erase_node(node1)
                     self.graph.erase_node(node2)
 
+    def construct_python_bytecode(self, tx, codegen: PyCodegen, stack_values):
+        if torch._dynamo.config.reorder_logs and len(tx.debug_locals) != 0:
+            for debug_var, args in tx.debug_locals:
+                codegen(debug_var)
+                for arg in args:
+                    codegen(arg)
+                codegen.extend_output(create_call_function(len(args), True))
+
+        self.side_effects.codegen_hooks(codegen)
+        self.side_effects.codegen_save_tempvars(codegen)
+        codegen.restore_stack(stack_values, value_from_source=not tx.export)
+        self.side_effects.codegen_update_mutated(codegen)
+
```

Comment:
I think we should move L998-L1003 down to after L1008, probably there are some variables refer to temp vars or mutated vars, though they should be basic types, but they can be arguments of these log functions. You can add unit tests for such cases as follow-ups.

Reply:
I can't add it all the way to the end, because I will add outputs to the codegen. So to my understanding this needs to go before the `codegen.restore_stack` call.